### ISSUE TITLE
Bug: Salt shows an error during the python-pip installation

### ISSUE THREAD

**Issue by porunov** [saltstack/salt#38857]
### Description of Issue/Question

I have a strange issue. When I install 'python-pip' library on CentOS 7.2 via salt state I get an error showing that `python-pip` failed to install/update. The strange thing is that it installs correctly but shows an error. I thought maybe it was because we have a symbol `-` in package's name but other packages with `-` are installed without errors. 

### Setup
(Please provide relevant configs and/or SLS files (Be sure to remove sensitive info).)

OS: CentOS 7.2

state:
```
pip_installed:
  pkg.installed:
    - name: python-pip
```

### Steps to Reproduce Issue
(Include debug logs if possible and relevant.)

1) Write a state file

2) Run on the master node:

`salt '*' state.apply -t 160`

3) See an error:
```
----------
          ID: pip_installed
    Function: pkg.installed
        Name: python-pip
      Result: False
     Comment: The following packages failed to install/update: python-pip
     Started: 22:35:38.426950
    Duration: 2211.963 ms
     Changes:
```

### Versions Report
(Provided by running `salt --versions-report`. Please also mention any differences in master/minion versions.)

```
Salt Version:
           Salt: 2016.11.1
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: 1.6.5
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: 1.5.0
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.5 (default, Nov  6 2016, 00:28:07)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.3.0
           RAET: 0.6.5
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4
 
System Versions:
           dist: centos 7.3.1611 Core
        machine: x86_64
        release: 3.10.0-327.4.5.el7.x86_64
         system: Linux
        version: CentOS Linux 7.3.1611 Core
```

**Comment by fredrikaverpil** [comment]
I have this exact issue.

**Comment by fredrikaverpil** [comment]
I have this exact issue.

I used to run CentOS 7.2.1511 just fine and now after having installed CentOS 7.3.1611 I started getting that issue, using the same sls which worked fine previously:

```yml
python-pip:
  pkg.installed:
    - require:
      - sls: centos.epel  # installs epel
```

All other pip packages which require `python-pip` fail as a result. Example:

```yml
requests:
  pip.installed:
    - require:
      - pkg: python-pip
```

**Comment by gtmanfred** [comment]
Epel also changed the name to python2-pip.

**Comment by terminalmage** [comment]
It does not. We used to try to provide suggestions when yum failed to find a match for a given package, but the prevailing opinion was that doing so slowed things down to an unacceptable level. Additionally, support for this relied on external dependencies (yum-utils for yum, and a dnf plugin for dnf) which themselves behaved inconsistently in different versions. For these reasons, we decided (I think for the 2016.3 release cycle) to do away with package name suggestions in yumpkg.py.
